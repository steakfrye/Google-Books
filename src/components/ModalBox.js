import React, { Component } from 'react';

import { evaluate, evaluateImage } from '../utils/evaluate';

// Displays book info when clicked
export class DetailBox extends Component {

  render() {
    const book = this.props.book.volumeInfo;

    return (
      <div>
        <div
          className="modal fade"
          id="exampleModal"
          tabIndex="-1"
          role="dialog"
          aria-hidden="true"
        >
          <div className="modal-dialog modal-lg" role="document">
            <div className="modal-content">
              <div className="modal-header">
                <h5 className="modal-title">
                  {book.title}, {evaluate(book.authors, 'authors')}
                  | {evaluate(book.title, 'title')}
                </h5>
                <button
                  type="button"
                  className="close"
                  data-dismiss="modal"
                  aria-label="Close"
                >
                  <span aria-hidden="true">&times;</span>
                </button>
              </div>
              <div className="modal-body">
                <div className="row">
                  <div className="image col-md-3">
                    <img
                      src={evaluateImage(book.imageLinks)}
                      alt={book.title}
                    />
                    <p className="font-weight-bold">
                      Genre: {evaluate(book.categories, 'genre')}
                    </p>
                    {book.ratingsCount > 0 ?
                      <p>Rating: {book.averageRating}/5</p> :
                      <p>No ratings yet.</p>
                    }
                  </div>
                  <div className="col-md-9">
                    <p>{evaluate(book.description, 'description')}</p>
                  </div>
                  <a href={book.infoLink}>View in Google Play store.</a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default DetailBox;
