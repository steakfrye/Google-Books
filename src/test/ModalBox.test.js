import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import ModalBox from '../components/ModalBox';

configure({ adapter: new Adapter() });

describe('<ModalBox />', () => {
  const book = {
    "id": "bZ_JqLD6NAEC",
    "volumeInfo": {
      "title": "Collaborative Quilting",
      "description": "A book about quilting.",
      "authors": [
       "Freddy Moran",
       "Gwen Marston"
     ],"publisher": "Sterling Publishing Company, Inc.",
     "imageLinks": {"thumbnail": "http://tinyurl.com/55"}
   }
 }
  const item = mount(<ModalBox book={book}/>)

  it('handles props', () => {
    expect(item.prop('book')).toEqual(book);
  });

  it('renders data from props', () => {
    const tree = renderer
      .create(item)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
