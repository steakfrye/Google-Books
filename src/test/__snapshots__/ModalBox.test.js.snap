// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalBox /> renders data from props 1`] = `
<div>
  <div
    aria-hidden="true"
    className="modal fade"
    id="exampleModal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog modal-lg"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
          >
            Collaborative Quilting
            , 
            Freddy Moran
            Gwen Marston
            | 
            Collaborative Quilting
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="row"
          >
            <div
              className="image col-md-3"
            >
              <img
                alt="Collaborative Quilting"
                src="http://tinyurl.com/55"
              />
              <p
                className="font-weight-bold"
              >
                Genre: 
                No genre found
              </p>
              <p>
                No ratings yet.
              </p>
            </div>
            <div
              className="col-md-9"
            >
              <p>
                A book about quilting.
              </p>
            </div>
            <a>
              View in Google Play store.
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
